src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1144,7 +1131,8 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
                         SR.net_http_client_execution_error,
                         WinHttpException.CreateExceptionUsingLastError());
                 }
-
+                requestHandle.SetParentHandle(connectHandle);
```

Comment:
I will add moving to the Strategy pattern as consideration in #3000.

WinHttpHandler is more complicated than ClientWebSockets because it ends up with additional objects like WinHttpResponseStream that can outlive the WinHttpHandler as well.